lib/screens/home/widget/task_tile.dart

```diff
@@ -1,71 +1,65 @@
-// ignore_for_file: public_member_api_docs, sort_constructors_first
-import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_bounce/flutter_bounce.dart';
 import 'package:flutter_svg/svg.dart';
+import 'package:todo_app/utils/show_overlay.dart';
 
-import '../../../Themes/colors.dart';
 import '../../../constants/spaces.dart';
+import '../../../services/data_deletion.dart';
 import '../../../services/data_update.dart';
-import '../../../utils/assets.dart';
+import '../../../themes/colors.dart';
+import '../../../constants/assets.dart';
+import '../../../utils/overlay_manager.dart';
 
-class FiAs3AmnaTaskTile extends StatefulWidget {
+class TaskTile extends StatelessWidget {
   final String text;
-  String id;
-  bool hasCompleted;
-  FiAs3AmnaTaskTile({
+  final String id;
+  final bool hasCompleted;
+
+  const TaskTile({
     Key? key,
     required this.text,
     this.id = "",
     required this.hasCompleted,
   }) : super(key: key);
 
-  @override
-  State<FiAs3AmnaTaskTile> createState() => _FiAs3AmnaTaskTileState();
-}
+  void _handleCheckBoxTap(BuildContext context) {
+    updateCompletedField(id, !hasCompleted);
+  }
+
+  void _handleDeleteTap() {
+    deleteData(id);
+  }
+
+  void _handleTileTap(BuildContext context) {
+    OverlayManager.storeContext(context);
+    ShowOverlay.show(
+      context,
+      initialTaskText: text,
+      isEditing: true,
+      id: id,
+    );
+  }
 
-class _FiAs3AmnaTaskTileState extends State<FiAs3AmnaTaskTile> {
   @override
   Widget build(BuildContext context) {
-    return GestureDetector(
-      onTap: () async {
-        await FirebaseFirestore.instance
-            .collection('todos')
-            .doc(widget.id)
-            .get()
-            .then(
-          (DocumentSnapshot documentSnapshot) {
-            if (documentSnapshot.exists) {
-              debugPrint(documentSnapshot['taskContent']);
-            }
-          },
-        );
-
-        //document snapshot and id is printed and overlay opens
-      },
+    return Bounce(
+      onPressed: () => _handleTileTap(context),
+      duration: const Duration(milliseconds: 100),
       child: Container(
-        margin: const EdgeInsets.only(
-          left: 20,
-          right: 20,
-          bottom: 10,
-        ),
-        padding:
-            const EdgeInsets.only(left: 25, top: 30, bottom: 30, right: 25),
+        margin: const EdgeInsets.only(left: 20, right: 20, bottom: 10),
+        padding: const EdgeInsets.symmetric(horizontal: 25, vertical: 30),
         decoration: BoxDecoration(
           color: CustomColors.tileColor,
```

Comment:
whenever you name colors instead of making them widget specific make them app specific,
primary, accent etc